os.java

```diff
@@ -37,29 +37,18 @@ public static void Crint (int []a, int []p)   {
 		sos.siodrum(pcb.jobNumber, pcb.jobSize, startingAddress, 0); 
 		RunOSTasks(a, p);
 	}
-	
-	
-	
+
 	public static void Dskint (int []a, int []p)  {
 		System.out.println("Disk Interrupt");
 		BookKeeping();
-		
-		 while(ioQueue!=null){
 			 PCB pcb = ioQueue.poll();
-			 System.out.println("Job #" + pcb.jobNumber + " doing I/O ");
-			 
-			 
-			 sos.siodisk(pcb.jobNumber);//  job doing I/O
-			 ioQueue.remove(pcb);  //remove i/o job from I/O Queue			 
-			 readyQueue.add(pcb.jobNumber);  // put job completed I/O to the readyQueue
+			 System.out.println("Job #" + pcb.jobNumber + " finished doing I/O ");
+			 pcb.status = "READY";
+			 readyQueue.add(pcb);  // put job completed I/O to the readyQueue
 			 
-		 }
 		RunOSTasks(a, p);
 	}
```

Comment:
The purpose of the diskInterrupt is to let the os know that i/o has completed.. You don't want to do sos.siodisk again here - just mark at as ready